contxt/models/events.py

```diff
@@ -109,36 +109,16 @@ class Event(ApiObject):
     deleted_at: Optional[datetime] = None
 
 
-@dataclass
-class TriggeredEventData(ApiObject):
-    _api_fields: ClassVar = (
-        ApiField("source_id"),
-        ApiField("subchain_triggered_event_id"),
-        ApiField("trigger_start_at", data_type=Parsers.datetime),
-        ApiField("trigger_end_at", data_type=Parsers.datetime),
-    )
-
-    source_id: str
-    subchain_triggered_event_id: str
-    trigger_start_at: datetime
-    trigger_end_at: datetime
-
-
 @dataclass
 class TriggeredEvent(ApiObject):
     _api_fields: ClassVar = (
         ApiField("id"),
         ApiField("event_id", creatable=True),
-        ApiField("Event", data_type=Event, attr_key="event"),
+        ApiField("Event", data_type=Event, attr_key="event", optional=True),
         ApiField("owner_id"),
-        ApiField("Owner", data_type=dict, attr_key="owner"),
+        ApiField("Owner", data_type=dict, attr_key="owner", optional=True),
         ApiField("is_public", data_type=bool),
-        ApiField(
-            "data",
-            data_type=lambda o: [TriggeredEventData.from_api(d) for d in loads(o)],
-            creatable=True,
-            updatable=True,
-        ),
+        ApiField("data", creatable=True, updatable=True),
```

Comment:
what about loading the json string? would depend on the exact shape, is it just a json object stored as a string?

```suggestion
        ApiField("data", data_type: lambda s: loads(s), creatable=True, updatable=True),
```

Reply:
The database column is a str, and there seems to be a mixture of json and strings present. Running `loads("test")` gave me an error: `json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`

Reply:
gotcha, my suggestion would be to solve for the common case (whatever that may be). for example, if the common case is json then maybe adding a simple helper property to parse it for you:

```python
    @cachedproperty
    def parsed_data(self) -> Optional[Any]:
        """Parse `data` as `json`, falling back to `str`"""

        if not self.data:
            return self.data

        # HACK: infer datatype (values seem to be mostly json or simple string)
        try:
            data_json = loads(self.data)
        except json.JSONDecodeError as e:
            return self.data
```

only if that's helpful, your call

Reply:
we can add this convenience later